feat: adding EventListingCard

jxmoose opened the pull request:
[//]: # "Feel free to customize this template to your project's vibes"

## :bread: What's new in this PR
EventListingCard component
activeEvents page

### :croissant: Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
added new fetch for active events. added event listing card component and a page to display it as well that supports horizontal scrolling


### :baguette_bread: Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

<img width="392" alt="Screenshot 2024-10-23 at 12 01 57 PM" src="https://github.com/user-attachments/assets/0f67aa01-319c-47d8-9e8d-e64790ef5299">



## :sandwich: How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

just look at localhost:3000/activeEvents to see the view

## :pie: Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."



## :pancakes: Relevant links
### :pretzel: Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'



### :bagel: Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"



CC: @celinechoiii

celinechoiii commented:
i'm not sure if you are still working on this but can you also include the search bar, menu icon, and adjust the background color? no functionality needed btw!

jxmoose committed:
feat: discover page

added new event components and page that shows all the active events in teh events database
